### PR TITLE
AK+LibWeb: Some porting of URL away from DeprecatedString

### PR DESCRIPTION
**Pull request by shannonbooth**
There is still plenty of porting left to do, but this is a good step forward :^)

Noticed some other fixes / tidy up along the way with regard to percent encode/decode in particular.

### CODE REVIEW

File: Userland/Libraries/LibWeb/HTML/HTMLHyperlinkElementUtils.cpp [hunk 1]
```diff
@@ -114,7 +114,7 @@ void HTMLHyperlinkElementUtils::set_username(DeprecatedString username)
         return;
 
     // 4. Set the username given this’s URL and the given value.
-    url->set_username(AK::URL::percent_encode(username, AK::URL::PercentEncodeSet::Userinfo));
+    url->set_username(username);
```
Comment (ADKaster) [inline Userland/Libraries/LibWeb/HTML/HTMLHyperlinkElementUtils.cpp]: The commit message here says "URL::set_password will apply a percent encode" but the commit is only updating `set_username`. Is there some behind the scenes trickery, or is that a typo?

File: Userland/Services/LaunchServer/Launcher.cpp [hunk 2]
```diff
@@ -393,14 +393,16 @@ bool Launcher::open_file_url(const URL& url)
...
-    auto parameters = url.query().split('&');
-    for (auto const& parameter : parameters) {
-        auto pair = parameter.split('=');
-        if (pair.size() == 2 && pair[0] == "line_number") {
-            auto line = pair[1].to_int();
-            if (line.has_value())
-                // TextEditor uses file:line:col to open a file at a specific line number
-                filepath = DeprecatedString::formatted("{}:{}", filepath, line.value());
+    if (url.query().has_value()) {
+        auto parameters = url.query()->bytes_as_string_view().split_view('&');
```
Comment (ADKaster) [inline Userland/Services/LaunchServer/Launcher.cpp:397]: you could also call `for_each_split_view` here to further allocate less. (split_view returns a Vector :) )

File: Userland/Libraries/LibWeb/HTML/HTMLHyperlinkElementUtils.cpp (hunk 1 above)
Comment (shannonbooth) [reply Userland/Libraries/LibWeb/HTML/HTMLHyperlinkElementUtils.cpp]: Right, that's a typo, thanks!

File: Userland/Services/LaunchServer/Launcher.cpp (hunk 2 above)
Comment (shannonbooth) [reply Userland/Services/LaunchServer/Launcher.cpp]: nice, wasn't aware of that function

**Review by ADKaster — APPROVED** [review]

**Pull request merged**